apps/src/javalab/backpack.module.scss

```diff
@@ -0,0 +1,19 @@
+@import 'color.scss';
+
+.fileListItem {
+  display: flex;
+  flex-direction: row;
+  box-sizing: border-box;
+  padding: 5px 15px;
+  width: 100%;
+
+  &:hover {
+    background-color: $background_gray;
+  }
+}
+
+.fileListItemDark {
+  &:hover {
+    background-color: $dark_charcoal;
+  }
```

Comment:
there's an SCSS convention for this type of styling: https://www.sassmeister.com/gist/be23c99595ee7191930c558e23e1e282

Reply:
Thanks! Do you know what this convention is called?

Reply:
no, i'm struggling to find a definitive name for it. in my head, i refer to it as "concatenating class names in SCSS" 😆

Reply:
I didn't grok the ampersand as literally concatenating whatever is provided, I think I get it now 🤦‍♂️

Reply:
Question about this convention -- I feel like it sort of implies that applying the `fileListItemDark` class will give you all the styling for a dark file list item, when really you need to apply both `fileListItem` and `fileListItemDark` (I think). a) is that right, and b) does it feel at all confusing to you?

Reply:
Context is I'm trying to do something similar in JavalabButton, and starting down the path of using mixins instead of two classes

Reply:
a) is that right - AFAIK, yes. there might be a better way to do this that i'm not seeing, though. if you don't want to apply both classes, you can use `@extend` (or a mixin like you suggested, but if we're always extending the class' styles, that's what `@extend` is for):
```scss
.fileListItem {
    &:hover {
        background-color: gray;
    }
    
    &Dark {
        &:hover {
            @extend .fileListItem;
            color: black;   
        }
    }
}
```

b) does it feel at all confusing to you? - yeah, it's not very intuitive. i would've assumed that the inner class automatically extends the outer class if i hadn't used the Sassmeister tool to make an example